src/Core/src/Core/IVisualDiagnosticsLayer.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Maui.Graphics;
+
+#if ANDROID || IOS
+using Microsoft.Maui.Graphics.Native;
+#endif
+
+namespace Microsoft.Maui
+{
+	public interface IVisualDiagnosticsLayer
+	{
+		/// <summary>
+		/// Gets or sets a value indicating whether to disable UI Touch Event Passthrough.
+		/// Enable this when you want to enable hit testing the current visual diagnostics layer without
+		/// interfacing with the underlaying UI.
+		/// </summary>
+		bool DisableUITouchEventPassthrough { get; set; }
+
+		/// <summary>
+		/// Gets or sets a value indicating whether to automatically scroll to an element when adding an adorner, if available.
+		/// </summary>
+		bool AutoScrollToElement { get; set; }
```

Comment:
If you click on an item in the LVT and we want to add an adorner, we can scroll directly to the item if it's contained within a scrollview. It can be useful if an item isn't on screen to go directly to it. I believe this was already a thing that existed for the other Hot Reload implementations, (CC @chabiss)

We're calling the MAUI Scrollview implementations for handling scrolling to the element, which those call onto native functions, including for WinUI. In this case, I'm not writing native implementations myself. It should work with all platforms. Considering we're relying on the MAUI implementation, I think it should be safe for us to have it. If the scroll view API here has issues, then it would need to be fixed at the MAUI level here, so it's not something we would need to handle ourselves.